internal/controller/numaresourcesscheduler_controller.go

```diff
@@ -186,6 +197,11 @@ func (r *NUMAResourcesSchedulerReconciler) syncNUMASchedulerResources(ctx contex
 
 	schedSpec := instance.Spec.Normalize()
 	cacheResyncPeriod := unpackAPIResyncPeriod(schedSpec.CacheResyncPeriod)
+	replicas, err := r.computeSchedulerReplicas(ctx, schedSpec)
+	if err != nil {
+		return nropv1.NUMAResourcesSchedulerStatus{}, fmt.Errorf("failed to compute scheduler replicas: %w", err)
```

Comment:
the benefit of the old approach was that the computation was done sporadically, so it was unlikely to fail.
I'm wondering if we should cache the value or if we should recompute every time. Ideally we should only recompute 1. when the user asks so or 2. when the cluster changes.

Reply:
In general listing all objects of some kind in every reconciliation step makes me uncomfortable (and yes, we do that already, and yes, I'm already uncomfortable).
In this case I expect us to reconcile if the object changed though, which should be still OK.
Don't we need to watch on nodes though? should we react if nodes are automatically added or deleted?

let's take a look about other operators (scheduler operator?) do. As much as I like this concept, I would avoid to be smarter than other standard operators.

Reply:
I can look for an option to enter this flow only if the reconciliation triggered by event which is related to node objects.
But I don't think this is how operators are suppose to work

Reply:
>last but not least: how we envision the autodection on hypershift? let's try hard to offer a consistent experience on hypershift and openshift

Yes, I will dedicate a separate PR for that.

Reply:
> I can look for an option to enter this flow only if the reconciliation triggered by event which is related to node objects.
> But I don't think this is how operators are suppose to work

it depends on how we define the autosizing feature. Let's first review how similar operator works.

Reply:
> In general listing all objects of some kind in every reconciliation step makes me uncomfortable (and yes, we do that already, and yes, I'm already uncomfortable).

Since we're using the controllerruntime framework, we should not be worried about redundant `List` calls, because the controllerruntime client first reads from cache so it doesn't actually performs an API call to the server:
https://github.com/openshift-kni/numaresources-operator/blob/b72a1b849f681d7d3e95cc45889928363a38aa3a/vendor/sigs.k8s.io/controller-runtime/pkg/client/client.go#L361
We are setting the cache here:
https://github.com/openshift-kni/numaresources-operator/blob/860724c25dae93a091905e643c9170f760da7ba0/cmd/main.go#L253

Reply:
right, this is probably good enough in this case.